compiler/rustc_mir_dataflow/src/framework/direction.rs

```diff
@@ -322,7 +322,8 @@ where
     fn apply(&mut self, mut apply_edge_effect: impl FnMut(&mut D, SwitchIntTarget)) {
         assert!(!self.effects_applied);
 
-        let targets = self.values.iter().map(|&value| SwitchIntTarget { value, target: self.bb });
+        let values = &self.body.switch_sources()[self.bb][self.pred];
```

Comment:
I don't expect this to affect much, but since we're only interested in `self.pred`, it could be cheaper to just look at the terminator of basic block with id `self.pred` and iterate over all targets filtering for ones for `self.bb`. Caching this map does not seem very useful if "computing" it for specific blocks is just a lookup and an iteration.

Reply:
Considering that this is the only user of `switch_sources`, I'm not sure it is carrying its weight.

Reply:
According to the discussion on the original pull request #95120, switch sources were introduced to avoid quadratic time complexity associated with the implementation strategy you are just suggesting.

Reply:
I guess if we have a switch with lots of targets... 🤷 added to my todo to do some experiments in the future